### DISCUSSION TITLE
REPORTING about failures of "BLT versions" of "dgtaheno" Firmware Marlin 2.1.2.1 on Artillery Genius

### DISCUSSION THREAD

**Discussion by fjherna** [dgtaheno/artillery-Genius#28] [General]
Hi All,

I have installed an Antclabs BLTouch device in my Artillery Genius.

Thanks a lot "dgtaheno" for sharing with the community your project. It is really interesting, helpful and it learned me a lot about the processes of compiling Firmware. 

I tried all three available Firmware Marlin-2.1.2.1 files from "dgtaheno": Marlin 2.1.2.1_BLT.hex, Marlin 2.1.2.1_BLT.hex and 
Marlin 2.1.2.1_BLT_corrections_pending on validation.hex

Unfortunately I have to report that none of those .hex files work with my Printer.
Basically the problem is after the printer shows at the  TFT message: "Failed to enable Bed Leveling" and at the "Notifications" the messages: 
[¡ blue] Invalid mesh.
[! red]  Failed to enable Bed Level

Then I followed instructions to get rid off that message by:
Menu->Movement->Bed Level->ABL->Start

the result is always the same: during "Bed Leveling" process when the BLT is going to test the 21st point, it gets OUT OF THE BED.
Obviously  it fails with message: [!red] Probing Failed.

If anyone have a similar printer it would be helpful to share information.

Kind regards,

Javier

**Comment by piunch** [comment]
Hi,

I'm having the same problem. I think that the Bed size is misconfigured on the firmware configuration.h prior to compilation, as in my Genius it states that Y axis has a 230mm length.

I've already opened an issue with this behaviour.

Thank you for your  support on this.

Kind regards,
Jorge.

**Comment by fjherna** [comment]
Hi All,
Sometime since my last post.
I have an Artillery Genius (simple and old).
I had some issues with a new EXTRUDER and I have removed the new one and re-installed to the original Artillery Genius Extruder Assy (Direct Titan with Volcano HotEnd).
Additionally I installed a original Antclab BLTouch.
I took the opportunity to test the Firmware Marlin-2.1.2.1:   "Marlin 2.1.2.1_BLT.hex"
As it was requested to report about the results, here are them.
The results were satisfactory: good Bed Leveling test with BLT
Now the BLT runs all points without any problem (always inside the bed) and I could save the results at the EEPROM.

Still is pending some other files:  "Marlin 2.1.2.1_BLT-295C.hex", "Marlin  2.1.2.1_BLT_corrections_pending on validation.hex".
However I am not sure if this later Firmware is older or newer that the other  ("Marlin 2.1.2.1_BLT.hex")
Please, would any of you mind to confirm me which of the two is older? If the latest is newer I could try that Firmware too and report the results.

Kind regards,

Javier

**Reply by fjherna** [reply]
Hi All,
Some issues about files "Marlin 2.1.2.1_BLT-295C.hex" and "Marlin 2.1.2.1_BLT_corrections_pending on validation.hex".

I flash one after the other Firmware and (by chance!) I observed that the date and time that is shown about Firmware is exactly the same (obtained in both cases with M115 G-code from Pronterface. i.e.

FIRMWARE_NAME:Marlin 2.1.2.1 (Oct 14 2023 03:48:28) SOURCE_CODE_URL:github.com/MarlinFirmware/Marlin PROTOCOL_VERSION:1.0 MACHINE_TYPE:Genius EXTRUDER_COUNT:1 UUID:cede2a2f-41a2-4748-9b12-c55c62f367ff

FIRMWARE_NAME:Marlin 2.1.2.1 (Oct 14 2023 03:48:28) SOURCE_CODE_URL:github.com/MarlinFirmware/Marlin PROTOCOL_VERSION:1.0 MACHINE_TYPE:Genius EXTRUDER_COUNT:1 UUID:cede2a2f-41a2-4748-9b12-c55c62f367ff

I think that both files, beside different names, are the same.

Kind regards,

Javier
